toolkit/pallets/block-participation/src/tests.rs

```diff
@@ -0,0 +1,95 @@
+use crate::mock::*;
+use frame_support::inherent::{InherentData, ProvideInherent};
+use frame_system::Origin;
+use pretty_assertions::assert_eq;
+use sp_block_participation::BlockProductionData;
+use sp_block_participation::{Slot, INHERENT_IDENTIFIER};
+
+#[test]
+fn inherent_clears_production_log_prefix() {
+	new_test_ext().execute_with(|| {
+		let initial_log = (1..=20).map(|i| (Slot::from(i), i + 100)).collect();
+		let expected_log: Vec<_> = (11..=20).map(|i| (Slot::from(i), i + 100)).collect();
+		pallet_block_production_log::Log::<Test>::set(initial_log);
+
+		Payouts::note_processing(Origin::<Test>::None.into(), Slot::from(10))
+			.expect("Should succeed");
+
+		let log = pallet_block_production_log::Log::<Test>::get();
+		assert_eq!(log, expected_log);
+	})
+}
+
+#[test]
+fn creates_inherent_if_data_present() {
+	let mut inherent_data = InherentData::new();
+	inherent_data
+		.put_data(
+			INHERENT_IDENTIFIER,
+			&BlockProductionData::<u32, u64>::new(Slot::from(24), vec![]),
+		)
+		.unwrap();
+
+	let inherent = Payouts::create_inherent(&inherent_data).expect("Should crate an inherent");
+
+	assert_eq!(inherent, crate::Call::<Test>::note_processing { up_to_slot: Slot::from(24) })
+}
+
+#[test]
+fn check_fails_if_data_is_missing() {
+	let inherent_data = InherentData::new();
+	let inherent = crate::Call::<Test>::note_processing { up_to_slot: Slot::from(25) };
+
+	let result = Payouts::check_inherent(&inherent, &inherent_data);
+	assert_eq!(result, Err(sp_block_participation::InherentError::UnexpectedInherent))
```

Comment:
there are also `assert_ok` and `assert_err` macros, consider them:
```suggestion
	assert_err!(
		Payouts::check_inherent(&inherent, &inherent_data),
		sp_block_participation::InherentError::UnexpectedInherent,
	);
```

Reply:
I honestly think `pretty_assertions::assert_eq` does a better job than those specialized assertions